### PR TITLE
[11.0][ADD] module helpdesk_mgmt_solution

### PR DESCRIPTION
**Pull request by yweng8111**
We have made some extensions for the helpdesk tickets. This is the first module that we want to publish. With this module the helpdesk user can create solutions for and assign them to helpdesk tickets.

### CODE REVIEW

**Comment by yweng8111** [comment]
I have got a build error. I'm not sure how to fix it. Can someone help me?

**Comment by dalonsod** [comment]
@yweng8111 see this in travis log:

imagen

**Comment by dalonsod** [comment]
Maybe a PR to fix it in v11 `helpdesk_type` addon is needed. Is that correct @DarioLodeiros ?

**Comment by yweng8111** [comment]
@dalonsod Thanks for your hints. Actually I have also seen this line. But I'm not sure if I should fix this in the same PR or just like you said to create a new PR.

**Comment by dalonsod** [comment]
OCA convention says that's better a PR per addon. 

Anyway, as `helpdesk_type` was successfully added to respository before, maybe there'a an issue in repo configuration for v11 now

File: helpdesk_mgmt_solution/wizard/solution.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        self.ensure_one()
+        solution_id = self.env["helpdesk.solution"].create(
+            self._prepare_solution_values()
+        )
+        self.ticket_id.solution_ids = [(4, solution_id.id)]
+
+    def add_solution_to_ticket(self):
+        self.ensure_one()
+        ticket_id = self._context.get("helpdesk_ticket_id")
+        if ticket_id:
```
Comment (chandni299) [inline helpdesk_mgmt_solution/wizard/solution.py]: Can we manage one more condition here, if we could not find "ticket_id" (let us say the user try to add a solution from the list view of Solutions directly, Helpdesk -> Configuration -> Solutions )
Same for "Remove Solution"

**Review by chandni299 — COMMENTED** [review]
Code Reviewed. All seems Good :+1:

File: helpdesk_mgmt_solution/wizard/solution.py (hunk 1 above)
Comment (yweng8111) [reply helpdesk_mgmt_solution/wizard/solution.py]: @chandni299 I have made some improvement in the wizard. Now we can set the ticket_id manually, if the context doesn't contain the "default_ticket_id".

**Review by benbrich — APPROVED** [review]
Functional review legit!

File: helpdesk_mgmt_solution/wizard/solution.py (hunk 1 above)
Comment (chandni299) [reply helpdesk_mgmt_solution/wizard/solution.py]: looks Good :+1:

**Review by chandni299 — APPROVED** [review]

**Comment by dreispt** [comment]
Travis is red.

**Comment by yweng8111** [comment]
> Travis is red.

Hi @dreispt,

That test_pylint error doesn't come from our module but from helpdesk_type. May I just change it in the same PR?

**Comment by dreispt** [comment]
@yweng8111 I proposed a fix here: https://github.com/OCA/helpdesk/pull/277
After merge your PR needs a rebase, so that it includes this fix.

**Comment by yweng8111** [comment]
I've rebased my branch.
@dreispt Please set the"no stale" label

**Pull request merged**